### PR TITLE
[Graphbolt] Add BasicFeatureStore.

### PR DESCRIPTION
**Pull request by frozenbugs**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by mfbalin — APPROVED** [review]
LGTM

File: tests/python/pytorch/graphbolt/impl/test_basic_feature_store.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        feature_store.read("node", "paper", "a"), torch.tensor([3, 2, 1])
+    )
+    assert torch.equal(
+        feature_store.read("edge", "paper-cites-paper", "b"),
+        torch.tensor([2, 5, 3]),
+    )
+    assert torch.equal(
+        feature_store.read("node", "paper", "a", torch.tensor([0, 1])),
+        torch.tensor([3, 2]),
+    )
```
[the remark this answers is not in the document]
Comment (frozenbugs) [reply tests/python/pytorch/graphbolt/impl/test_basic_feature_store.py:26]: Sorry, I don't get it, do you mean add `feature_stores = None` in the end of the test? Why do we need to do so?

[the remark this answers is not in the document]
Comment (mfbalin) [reply tests/python/pytorch/graphbolt/impl/test_basic_feature_store.py:26]: It is not me who wrote this code but from my understanding, when we use files, we gotta destroy references properly by setting feature store to None

**Review by Rhett-Ying — APPROVED** [review]

File: tests/python/pytorch/graphbolt/impl/test_basic_feature_store.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (frozenbugs) [reply tests/python/pytorch/graphbolt/impl/test_basic_feature_store.py:26]: Yes, but I did not file here.

**Pull request merged**